src/Trace/TraceClient.php

```diff
@@ -0,0 +1,177 @@
+     *
+     * @param Trace[] $traces The trace logs to send.
+     * @return bool
+     * @throws ServiceException
+     */
+    public function insertBatch(array $traces)
+    {
+        // throws ServiceException on failure
+        $this->connection->patchTraces([
+            'projectId' => $this->projectId,
+            'traces' => array_map(function ($trace) {
+                return $trace->info();
+            }, $traces)
+        ]);
+        return true;
+    }
+
+    /**
+     * Lazily find or instantiates a trace. There are no network requests made at this
+     * point. To see the operations that can be performed on a trace please
+     * see {@see Google\Cloud\Trace\Trace}.
+
+     * @param  string $traceId [optional] The trace id of the trace to reference.
+     * @return Trace
+     */
+    public function trace($traceId = null)
+    {
+        return new Trace($this->connection, $this->projectId, $traceId);
+    }
+
+    /**
+     * Fetch all traces in the project
+     *
+     * @see https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces/list Trace list API documentation.
+     *
+     * @param array $options [optional] {
+     *      Configuration options.
+     *
+     *      @type string $viewType Type of data returned for traces in the list.
+     *            Can be one of 'VIEW_TYPE_UNSPECIFIED', 'MINIMAL', 'ROOTSPAN', or
+     *            'COMPLETE'
+     *      @type int $pageSize Maximum number of traces to return
+     *      @type string $pageToken Token identifying the page of results to return
+     *      @type string $startTime Start of the time interval during which trace data
+     *            was collected. This timestamp in nanoseconds should be in "Zulu" format.
+     *            Example: '2014-10-02T15:01:23.045123456Z'
+     *      @type string $endTime End of the time interval during which trace data was
+     *            collected. This timestamp in nanoseconds should be in "Zulu" format.
+     *            Example: '2014-10-02T15:01:23.045123456Z'
+     *      @type string $filter An optional filter for the request
+     *      @type string $orderBy Field used to sort the returned traces. Can be one
+     *            of 'traceId', 'name', 'duration', 'start'. Descending order can be
+     *            specified by appending 'desc' to the sort field (for example,
+     *            'name desc'). Only one sort field is permitted.
+     * }
+     * @return ItemIterator<Trace>
+     */
+    public function traces(array $options = [])
+    {
+        return new ItemIterator(
+            new PageIterator(
+                function (array $trace) {
+                    $trace += ['spans' => null];
+                    return new Trace($this->connection, $trace['projectId'], $trace['traceId'], $trace['spans']);
+                },
+                [$this->connection, 'listTraces'],
+                $options + ['projectId' => $this->projectId],
```

Comment:
Switch the order of these (`['projectid' => $this->projectId] + $options`) to ensure that `projectId` is set to the class value rather than one given in the method options.